core/src/com/unciv/models/ruleset/Ruleset.kt

```diff
@@ -575,9 +575,18 @@ class Ruleset {
             if (building.requiredBuildingInAllCities != null)
                 lines.add("${building.name} contains 'requiredBuildingInAllCities' - please convert to a \"" +
                         UniqueType.RequiresBuildingInAllCities.text.fillPlaceholders(building.requiredBuildingInAllCities!!)+"\" unique", RulesetErrorSeverity.Warning)
-            for (unique in building.uniqueObjects)
+            for (unique in building.uniqueObjects) {
                 if (unique.placeholderText == "Creates a [] improvement on a specific tile" && !tileImprovements.containsKey(unique.params[0]))
                     lines += "${building.name} creates a ${unique.params[0]} improvement which does not exist!"
+                if (unique.placeholderText == UniqueType.RequiresAnotherBuilding.placeholderText
+                        || unique.placeholderText == UniqueType.RequiresBuildingInAllCities.placeholderText
+                        || unique.placeholderText == UniqueType.RequiresBuildingInSomeCities.placeholderText
+                ) {
+                    val buildingName = unique.params[0]
+                    if (!buildings.containsKey(buildingName))
+                        lines += "${building.name} requires $buildingName which does not exist!"
+                }
+            }
```

Comment:
This is unnecessary, due to the checkUniques function below, which will do this anyway

Reply:
OK, I delete it.